src/lib/format/partial/MessageFormatter.ts

```diff
@@ -233,6 +233,10 @@ export namespace MessageFormatter {
                 }
                 fieldData.fieldObjectType = fieldObjectType;
                 fieldData.canBeUndefined = canBeUndefined;
+            } else if (fieldData.isRepeatField) {
+                if (fieldData.type === ENUM_TYPE) {
+                    fieldData.fieldObjectType = `keyof typeof ${fieldData.fieldObjectType}`;
+                }
```

Comment:
Previously, we only append the `keyof typeof` when `isRepeatField` is false (see the above if condition)

Reply:
Did you test this? The way it combines needs parens or `Array<...>` to work accurately, because `keyof typeof Foo[]` will give you keys of Array, rathe than an array of keys.

Reply:
yeah, I checked the `message.hbs` file, the template is doing something like if `isRepeated`, then do `Array<{fieldObjectType}>`, so should be fine

A local test is ongoing

Reply:
Test finished, can confirm the local generated web-api has correct grpc types